src/main/xerus/monstercat/downloader/Download.kt

```diff
@@ -147,3 +152,7 @@ class ReleaseDownload(private val release: Release, private var tracks: Collecti
 }
 
 class EmptyResponseException(term: String) : Exception("No file found for $term!")
+
+class WrongResponseTypeException(term: String, mime: String) : Exception("Wrong MIME type returned for $term: type $mime unexpected!")
+
+class WrongResponseCodeException(term: String, code: Int) : Exception("Response code is not \"200 OK\" for $term : is $code")
```

Comment:
I'd reword this to make it more understandable to the average user, something like this:
```suggestion
class WrongResponseCodeException(term: String, code: Int) : Exception("Error downloading $term: Server returned code $code")
```
and maybe include the full status line, not just the code

Reply:
Sure, I'll add the status text as well.

Reply:
6c9dd3d3da212c2e79512595a7811052f37bc0f9

Reply:
6c9dd3d3da212c2e79512595a7811052f37bc0f9